classes/Search.php

```diff
@@ -220,5 +225,13 @@ private function set_page_title($data) {
         $DATA->set_page_metadata($this_page, 'title', $pagetitle);
     }
 
+    private function prettifySearchString($string) {
+        if ( preg_match('/speaker:(\d+)/', $string, $m) ) {
+            $MEMBER = new MEMBER(array('person_id'=>$m[1]));
+            $name = 'speaker:' . $MEMBER->full_name();
+            $string= str_replace("speaker:$m[1]", $name, $string);
```

Comment:
Ooh, or I see the sidebar and the title do it using query_description_long (via variables email_desc and searchdescription) – perhaps an opportunity to factor this so it's only in one place?